x-pack/plugins/apm/public/hooks/useServiceMetricCharts.ts

```diff
@@ -4,53 +4,28 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { useMemo } from 'react';
-import { MetricsChartAPIResponse } from '../../server/lib/metrics/get_all_metrics_chart_data';
-import { MemoryChartAPIResponse } from '../../server/lib/metrics/get_memory_chart_data';
-import { loadMetricsChartDataForService } from '../services/rest/apm/metrics';
-import { getCPUSeries, getMemorySeries } from '../selectors/chartSelectors';
+import { MetricsChartsByAgentAPIResponse } from '../../server/lib/metrics/get_metrics_chart_data_by_agent';
+import { loadMetricsChartData } from '../services/rest/apm/metrics';
 import { IUrlParams } from '../context/UrlParamsContext/types';
 import { useFetcher } from './useFetcher';
 
-const memory: MemoryChartAPIResponse = {
-  series: {
-    memoryUsedAvg: [],
-    memoryUsedMax: []
-  },
-  overallValues: {
-    memoryUsedAvg: null,
-    memoryUsedMax: null
-  },
-  totalHits: 0
-};
-
-const INITIAL_DATA: MetricsChartAPIResponse = {
-  memory,
-  cpu: {
-    series: {
-      systemCPUAverage: [],
-      systemCPUMax: [],
-      processCPUAverage: [],
-      processCPUMax: []
-    },
-    overallValues: {
-      systemCPUAverage: null,
-      systemCPUMax: null,
-      processCPUAverage: null,
-      processCPUMax: null
-    },
-    totalHits: 0
-  }
+const INITIAL_DATA: MetricsChartsByAgentAPIResponse = {
+  charts: []
 };
 
 export function useServiceMetricCharts(urlParams: IUrlParams) {
   const { serviceName, start, end, kuery } = urlParams;
 
-  const { data = INITIAL_DATA, error, status } = useFetcher(
+  const { data = INITIAL_DATA, error, status } = useFetcher<
+    MetricsChartsByAgentAPIResponse
+  >(
     () => {
       if (serviceName && start && end) {
-        return loadMetricsChartDataForService({
+        return loadMetricsChartData({
           serviceName,
+          // TODO: Remove hard-coded agent -- should we add agent name to url params or query for
+          // agent based on service name in the back end to make this simpler in the client?
+          agentName: serviceName === 'opbeans-java' ? 'java' : 'default',
```

Comment:
I'm not sure if it makes more sense to use the agentName by figuring it out in the client and including it in this request or if it makes more sense to query just by service name and then have the back end look up what agent that service is using? Either way, we obviously need to remove this hard-code before merging :)

Reply:
I opted to use the agentName that we have in the client already since it already comes back to us when we query for service details.